From 75dc553f10df3cf48227ec69b9465009eebf7b8b Mon Sep 17 00:00:00 2001
From: Georges Basile Stavracas Neto <georges.stavracas@gmail.com>
Date: Wed, 17 Jan 2018 20:05:03 -0200
Subject: [PATCH] common: Don't unconditionally define HAVE_WAYLAND

This header unconditionally defines HAVE_WAYLAND when GTK
is built with Wayland support. This breaks the build when
building with Wayland disabled.

https://bugzilla.gnome.org/show_bug.cgi?id=785414
(cherry picked from commit 62f07b2c6d8a28378e1f2c9f5d0c3241332ee8cf)
[Added HAVE_WAYLAND config.h definition for autotools to fix runtime]
Signed-off-by: Mart Raudsepp <leio@gentoo.org>
---
 configure.ac                       | 1 +
 panels/common/gnome-settings-bus.h | 2 --
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index f810bfb1f..24940a798 100644
--- a/configure.ac
+++ b/configure.ac
@@ -291,6 +291,7 @@ AS_IF([test "$enable_wayland" != "no"],
               [AC_DEFINE(HAVE_UDEV, 1, [System has udev])])
 
         PKG_CHECK_MODULES(WAYLAND, gdk-wayland-3.0)
+        AC_DEFINE(HAVE_WAYLAND, 1, [Compile with Wayland support])
 ])
 AM_CONDITIONAL(HAVE_WAYLAND, [test "x$enable_wayland" != "xno"])
 
diff --git a/panels/common/gnome-settings-bus.h b/panels/common/gnome-settings-bus.h
index ce58f5805..763a9cca7 100644
--- a/panels/common/gnome-settings-bus.h
+++ b/panels/common/gnome-settings-bus.h
@@ -5,8 +5,6 @@
 
 #ifdef GDK_WINDOWING_WAYLAND
 
-#define HAVE_WAYLAND 1
-
 static inline gboolean
 gnome_settings_is_wayland (void)
 {
-- 
2.17.0

